proxy/mempool/mempool_schedule.py

```diff
@@ -278,18 +289,44 @@ def chain_id(self) -> int:
     def _add_tx_to_sender_pool(self, sender_pool: MPSenderTxPool, tx: MPTxRequest) -> None:
         LOG.debug(f'Add tx {tx.sig} to mempool with {self.tx_cnt} txs')
 
+        if self._sender_pool_heartbeat_queue.find(sender_pool) is not None:
+            self._sender_pool_heartbeat_queue.pop(sender_pool)
+        
         sender_pool.add_tx(tx)
         self._tx_dict.add_tx(tx)
 
         # the first tx in the sender pool
         if sender_pool.len_tx_nonce_queue == 1:
```

Comment:
```
if is_new_pool: 
    self._sender_pool_dict[sender_pool.sender_address] = sender_pool
```

Reply:
I have been stuck on this for a while, and unless I'm wrong in my debugging, I've held up on committing these suggested changes until further advised...

I did these last two suggested changes, but they break the test `test_add_tx()`. I suspect because of the `None` addresses returned from `NeonAccount.create()`, the following call is returning an empty sender pool:

    sender_pool = self._get_or_create_sender_pool(tx.sender_address)

This would be the reason why your suggestion above with `is_new_pool = sender_pool.len_tx_nonce_queue == 0` is failing.

So I haven't committed the suggestion change as the current code works and is more defensive given the possible broken test.

Reply:
Ignore that last comment, I found the issue...

What I should be doing is popping from the heartbeat queue when the pool is empty. That fixed it. Will commit this soon.